docs/source/upgrade-2.0.rst

```diff
@@ -0,0 +1,187 @@
+=======================
+Upgrading to cocotb 2.0
+=======================
+
+***************
+``cocotb.fork``
+***************
+
+Change
+======
+
+:external+cocotb1.9:py:func:`cocotb.fork` was removed and replaced with :func:`cocotb.start_soon`.
+
+Rationale
+=========
+
+``cocotb.fork`` would turn :term:`coroutine`\ s in :class:`~cocotb.task.Task`\s that would run concurrently to the current :term:`task`.
+However, it would immediately run the coroutine until the first :keyword:`await` was seen.
+This made the scheduler re-entrant and caused a series of hard to diagnose bugs
+and required extra state/sanity checking leading to runtime overhead.
+For these reasons ``cocotb.fork()`` was deprecated in cocotb 1.7 and replaced with ``cocotb.start_soon()``.
+``cocotb.start_soon()`` does not start the coroutine immediately, but rather "soon",
+preventing scheduler re-entrancy and sidestepping an entire class of bugs and runtime overhead.
+
+How to Upgrade
+==============
+
+* Replace all instances of ``cocotb.fork()`` with ``cocotb.start_soon()``.
+* Run tests to check for any changes in behavior.
+
+Coroutines run immediately
```

Comment:
Feel free to copy (or link) content from https://fossi-foundation.org/blog/2021-10-20-cocotb-1-6-0; pictures will help a lot when describing the situation.